docs/proposals/ktl.md

```diff
@@ -0,0 +1,243 @@
+# ktl (kubectl2)
+
+## Metadata
+
+## Table of Contents
+
+- [Title](#title)
+   - [Metadata](#metadata)
+   - [Table of Contents](#table-of-contents)
+   - [Summary](#summary)
+   - [Motivation](#motivation)
+   - [Reference-level explanation](#reference-level-explanation)
+   - [Graduation Criteria](#graduation-criteria)
+   - [Alternatives](#alternatives-optional)
+   - [Unresolved Questions](#unresolved-questions-optional)
+
+## Summary
+
+- Stop trying to wholesale move kubectl out of kubernetes/kubernetes
+  - we have been working on this for a year with limited success
+- Instead, create a new cli command called `ktl`.
+  - `ktl` lives in its own repo and vendors commands developed in other repos (kubernetes/kubernetes, kubernetes/kubectl)
+- `ktl` is built and released frequently
+  - vendored sub commands follow their own release cycles and updated by released the vendored code
+
+## Motivation
+
+The core Kubernetes cli is published as a single binary called `kubectl`.
+`kubectl` is a statically linked go binary in the kubernetes/kubernetes
+repo that contains many sub commands to perform operations against a Kubernetes cluster, such as:
+
+- creating & updating objects defined by commandline args
+- managing objects using configuration files
+- debugging objects in a cluster
+
+Benefits of this approach:
+
+- having a single binary facilitates discovery for the suite of Kubernetes cli commands
+- static binary makes it simple to distribute the cli across all OS platforms
+- static binary makes it simple for users to install the cli
+- single binary limits the size of the install (MB) vs distributing ~50 separate go commands (GB)
+- centralizing the development facilitates the construction of shared libraries and infrastructure
+- leveraging the kubernetes/kubernetes repo infrastructure provides a process to build and release
+
+Challenges of this approach:
+
+- the cli cannot be released at a different intervals from the cluster itself
+  - slows down velocity
+  - slows down feedback on alpha and beta features - longer time to GA
+  - each release is heavier weight due to its length
+  - obfuscates version skew as a first class problem
```

Comment:
basically we must have a different release cycle between cli and cluster itself.